sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -307,19 +268,27 @@ def get_partition_properties(self, partition_id):
             )
         return output
 
-    def close(self):
+    def _close(self):
         # type:() -> None
         self._conn_manager.close_connection()
 
 
 class ConsumerProducerMixin(object):
+    def __init__(self):
+        self._name = ""
+        self._client = None
+        self._handler = None
+        self._loop = None
 
     def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+    def _create_handler(self, auth):
```

Comment:
See previous answer, a mixin is not an interface, by doing that you break the MRO

Reply:
Still needs to be addressed